contentcuration/contentcuration/frontend/shared/vuex/file/utils.js

```diff
@@ -61,6 +64,42 @@ export function storageUrl(checksum, file_format) {
   return `/content/storage/${checksum[0]}/${checksum[1]}/${checksum}.${file_format}`;
 }
 
+export async function getH5PMetadata(fileInput, metadata) {
+  // const file = fileInput.files[0];
+  // console.log(typeof(files), file);
+  const zip = new JSZip();
+  zip
+    .loadAsync(fileInput)
+    .then(function(zip) {
+      const h5pJson = zip.file('h5p.json');
+      // console.log(h5pJson);
+      if (h5pJson) {
+        return h5pJson.async('text');
+      } else {
+        throw new Error('h5p.json not found in the H5P file.');
+      }
+    })
+    .then(function(h5pContent) {
+      const data = JSON.parse(h5pContent);
+      if (Object.prototype.hasOwnProperty.call(data, 'title')) {
+        metadata.title = data['title'];
+      }
+      if (Object.prototype.hasOwnProperty.call(data, 'language') && data['language'] !== 'und') {
+        metadata.language = data['language'];
+      }
+      if (Object.prototype.hasOwnProperty.call(data, 'authors')) {
+        metadata.author = data['authors'];
+      }
+      if (Object.prototype.hasOwnProperty.call(data, 'license')) {
+        metadata.license = data['license'];
+      }
+    })
+    .catch(function(error) {
+      console.error('Error loading or extracting H5P file:', error);
```

Comment:
I think this is fine for now - we can file a follow up issue to better handle when H5P files are uploaded that do not have the required metadata, but it should be an edge case (hopefully!)